aws-sdk-core/lib/aws-sdk-core/cloudfront/url_signer.rb

```diff
@@ -0,0 +1,136 @@
+require 'base64'
+require 'uri'
+require 'time'
+require 'json'
+require 'openssl'
+
+module Aws
+  module CloudFront
+        # Allows you to create signed URLs for Amazon CloudFront resources
+        #
+        # Example Use:
+        # signer = Aws::CloudFront::UrlSigner.new
+        # url = signer.signed_url(
+        #	"CF_KEYPAIR_ID",
+        #	"/path/to/private/key/cf_private_key.pem"
+        # )
+        #
+        class UrlSigner
+        	# @key_pair_id [String] :ID of the key pair
+            # @private_key_path [String] :Path to the private key used for signing
+            def initialize(key_pair_id, private_key_path)
+                @key_pair_id = key_pair_id
+                @private_key = File.open(private_key_path, 'rb') { |f| f.read }
+            end
+
+            # create a signed Amazon CloudFront URL
+            # @param [String] URL to sign
+            # @option param [String | Integer] UTC Unix timestamp
+            # @option param [String] JSON Policy
+            def signed_url(url, expires = nil, policy = nil)
+                url_sections = url.split('://')
+                if url_sections.length < 2
+                    raise ArgumentError, "Invaild URL:#{url}"
+                end
+
+                scheme = url_sections[0].gsub('*', '')
+                uri = "#{scheme}://#{url_sections[1]}"
+                signature = signature(resource(scheme, uri), expires, policy)
+                start_flag = (uri.include? '?') ? '&' : '?'
+
+                uri = "#{uri}#{start_flag}#{signature}"
+                if scheme == 'rtmp'
+                    rtmp_url(URI(uri))
+        		else
+        			uri
+                end
+            end
+
```

Comment:
You should add a `private` statement here to keep these methods from being documented as part of the public interface.